app/controllers/api/v0/events_controller.rb

```diff
@@ -6,11 +6,11 @@ def create
     render(json: error, status: error.status_code) and return if error
 
     inbound_binding.events.each do |event|
-      # first, schema validate the event.data object
+      event.data[:uuid] = CompactUuid.pack(current_user_uuid) if current_user_uuid
 
-      event.data[:user_uuid] = current_user_uuid if current_user_uuid
+      avro_encoded_data = KafkaAvroTurf.instance.encode(event.data, schema_name: event.type)
```

Comment:
This is probably the schema version question... how do we deal with clients with an older schema than what is live in this backend?  Can clients communicate that they are on a certain version and then can we deal with that?

Reply:
good question.  we do need to pass a version (defaults to current version) if we need this capability.   In this case, we'd encode and decode w/ that version